bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/internal/DiscoveryServiceRegistryImpl.java

```diff
@@ -210,13 +212,19 @@ public synchronized void removeDiscoveryListener(DiscoveryListener listener) thr
     }
 
     @Override
-    public synchronized void thingDiscovered(DiscoveryService source, DiscoveryResult result) {
+    public synchronized void thingDiscovered(final DiscoveryService source, final DiscoveryResult result) {
         synchronized (cachedResults) {
             cachedResults.put(source, result);
         }
-        for (DiscoveryListener listener : this.listeners) {
+        for (final DiscoveryListener listener : this.listeners) {
             try {
-                listener.thingDiscovered(source, result);
+                AccessController.doPrivileged(new PrivilegedAction<Void>() {
+                    @Override
+                    public Void run() {
+                        listener.thingDiscovered(source, result);
```

Comment:
What doPrivileged does is, that it executes the code inside the call (listener.thingDiscovered(source, result);) with the permission of the calling bundle (in that case the core.discovery bundle). If the implementation of the listener again has a security context this security context is applied.